### ISSUE TITLE
Most of the time 2x fan test at startup

### ISSUE THREAD

**Issue by fiechr** [madworm/Youyue-858D-plus#2]
Hi!

It's not really a problem, because besides that everything works fine, but still:

Most of the time I switch on the station, it simply does the show firmware version and fan test part twice. So after the first fan test the display flickers briefly, shows the version still/again and the fan spins up again. After that the three dashes (or temperature) show and everything works as expected.
About 1/5th of the time the station does only one test.

Any ideas where to look?

I'm using the Youyue 858D+ 20140415 v6.0 board with the ATmega328p and the binary file which is kindly provided here. I also didn't change any settings yet - so everything is set to default.

This is how I flashed the chip (I renamed the file):

```
avrdude -c usbasp -p atmega328p -v -U lfuse:w:0xE2:m -U hfuse:w:0xDF:m -U efuse:w:0x05:m -U lock:w:0x0F:m

avrdude -c usbasp -p atmega328p -v -e -U flash:w:858d.hex:i
```

I never used avrdude before, so I post this too, just in case I did something wrong...

**Comment by madworm** [comment]
The FUSE settings look fine.

I have come across your issue a couple of times before. It seems that some AVR chips experience watchdog resets (which they shouldn't). My theory/speculation: bad chip (maybe factory rejects sold cheaply by non-reputable sources).

If you can compile the code yourself, try un-commenting the "//#define DISPLAY_MCUSR" option in "youyoue858d.h", recompile & reflash.

This will briefly show the contents of the MCUSR register (in decimal), showing the cause of reset.

// ATmega168 MCUSR (MSB to LSB): x-x-x-x-WDRF-BORF-EXTRF-PORF

If you get any decimal number with the 3rd bit set (WDRF) something might be wrong with your chip, e.g. the decimal number 8 would be a bad thing.

If you can't compile the firmware, let me know and I'll do it.

**Comment by madworm** [comment]
The quickest test might be using a different chip, if you have one. Preferably from a different source.

**Comment by fiechr** [comment]
Thank you for your fast response!

As I have no other chip available at the moment I will rebuild and check what you suggested (and report back).

I took this chip from an Arduino clone I don't need at the moment. But already ordered some more to replace it.

**Comment by fiechr** [comment]
I just rebuilt and flashed - everything went fine. Good thing I also added a ISP header - so I didn't even had to open the thing again. :-)

I switched the station about 10 times on/off, waiting a few seconds in between:

A normal start looks like this:

> 1. _switching ON_
> 2. 005
> 3. 1.43
> 4. _fan spins up (and stops after a while)_
> 5. ---

But most of the time I get instead:

> 1. _switching ON_
> 2. 005
> 3. 1.43
> 4. _fan spins up (and stops after a while)_
> 5. 000
> 6. 1.43
> 7. _fan spins up again (and stops after a while)_
> 8. ---

**Comment by madworm** [comment]
Now it's getting really weird.

By looking at the code, you can see that MCUSR is set to 0 after displaying it. In the abnormal case, it shows '000' the 2nd time, which means that the program counter must have changed backwards. But I don't see anything in the C-code that would do such a thing.

The chip also didn't do a proper hardware reset, as that would have written a new non-zero value to MCUSR. Basically it should NEVER display 0 and 8.

Please retest with a new chip!

**Comment by fiechr** [comment]
> Please retest with a new chip!

Ok, I will do that as soon as I get the new ones.

Thanks again!

**Comment by fiechr** [comment]
In the meantime I ordered two new 328p from Reichelt, because I needed something else from them anyway.

The result: After a few tests the problem really seems to be gone for good. I didn't get 000 anymore with the new chip. :-)

So the other one I pulled from the Arduino clone seems to be of less quality, although it always worked fine there. But then again I never did really complicated stuff with it.

I also ordered a second station for my other workbench. If I get my "chinese" ATmegas I ordered sooner, I will still test one of them first, before putting the one in I know now seems to be of genuine quality. Just for curiosity.
